assets/differentialabundance_report.Rmd

```diff
@@ -543,6 +546,31 @@ for (assay_type in rev(names(assay_data))){
 
       print(htmltools::tagList(do.call("plotly_scatterplot", plot_args)))
     }
+    percentVar_list[[assay_type]][[iv]] <- percentVar
+  }
+}
+```
+
+```{r, echo=FALSE, results='asis', eval=params$report_scree}
+cat(paste0("\n#### Scree plot {.tabset}"))
+cat(paste0("\nThe following scree plot visualizes what percentage of total variation in the data can be explained by each of the principal components computed.\n"))
+
+for (assay_type in names(percentVar_list)) {
+  for (iv in names(percentVar_list[[assay_type]])) {
```

Comment:
You don't need this loop- the scree plot is not impacted by the variable and it's not used for coloring or anything, so all you're doing is making duplicate plots. One per assay type will do.

Reply:
Oops, you are right!